### PR TITLE
[REF] mrp: bom not required on production order

### PR DESCRIPTION
**Pull request by sle-odoo**
task-2226404

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -801,7 +820,10 @@ def action_view_mrp_production_sources(self):
     def action_confirm(self):
         self._check_company()
         for production in self:
-            production.consumption = production.bom_id.consumption
+            if not production.bom_id:
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py:823]:
```python
            if production.bom_id:
                production.consumption = production.bom_id.consumption
            else:
                production.consumption = 'flexible'
```

File: addons/mrp/wizard/mrp_product_produce.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def default_get(self, fields):
             if 'qty_producing' in fields:
                 res['qty_producing'] = todo_quantity
             if 'consumption' in fields:
-                res['consumption'] = production.bom_id.consumption
+                res['consumption'] = production.bom_id.consumption if production.bom_id else 'flexible'
```
Comment (Whenrow) [inline addons/mrp/wizard/mrp_product_produce.py:42]: same in _workorders_create

File: addons/mrp/views/mrp_production_views.xml [hunk 3]
```diff
@@ -125,11 +126,11 @@
                                     <button type="action"
                                         name="%(mrp.action_change_production_qty)d"
                                         context="{'default_mo_id': id}"
-                                        string="Update" class="oe_link" attrs="{'invisible': ['|', ('state', 'in', ('draft', 'done','cancel')), ('id', '=', False)]}"/>
+                                        string="Update" class="oe_link" attrs="{'invisible': ['|', '|', ('state', 'in', ('draft', 'done','cancel')), ('id', '=', False), ('bom_id', '=', False)]}"/>
```
Comment (Whenrow) [inline addons/mrp/views/mrp_production_views.xml:129]: I would prefer adapt the change qty wizard :no_good:

Comment (sle-odoo) [reply addons/mrp/views/mrp_production_views.xml:129]: nope

File: addons/mrp/models/mrp_production.py (hunk 1 above)
Comment (sle-odoo) [reply addons/mrp/models/mrp_production.py:823]: wow wow wow
workorders without bom?
i'll ask @oma-odoo what's her opion on that.. but it stinks

**Comment by sle-odoo** [comment]
robodoo r+

**Pull request closed without merging**